File: Prototype/BigGp/src/Model/SmallNodes/Operators/Logic/CompareOperator.java
package Model.SmallNodes.Operators.Logic;

import Model.Node;
import Model.SmallNodes.Operators.Operator;
import Serialization.Token;

import java.util.ArrayList;

public class CompareOperator extends Operator {

    private final int BOUND;

    @Override
    public void mutate() {

        String newOperator = getSpecifiedOperator(getRandomInt(BOUND));

        while (newOperator.equals(getValue())) {
            newOperator = getSpecifiedOperator(getRandomInt(BOUND));
        }

        setOperator(newOperator);
    }

    private String getSpecifiedOperator(int operatorId) {
        String operatorValue = "";

        switch (operatorId) {
            case 0 -> operatorValue = " == ";
            case 1 -> operatorValue = " != ";
            case 2 -> operatorValue = " > ";
            case 3 -> operatorValue = " < ";
            case 4 -> operatorValue = " >= ";
            case 5 -> operatorValue = " <= ";
        }

        return operatorValue;
    }

    public CompareOperator(Node parentNode) {
        super(parentNode, "CompareOperator");
        BOUND = 6;

        setOperator(getSpecifiedOperator(getRandomInt(BOUND)));

    }

    public CompareOperator(Node parentNode, ArrayList<Token> tokens) {
        super(parentNode, tokens);
        BOUND = 6;
    }
}

File: Interactive.Genetic.Programming.Environment/src/Domain/Model/Nodes.Small/Operators.Logic/CompareOperator.cs
using Model.Abstract;
using Utils;

namespace Model.Nodes.Small.Operators.Logic;

public sealed class CompareOperator : Operator
{
    private const int Bound = 6;

    public override void Mutate()
    {
        var newOperator = GetSpecifiedOperator();

        while (newOperator == Value)
        {
            newOperator = GetSpecifiedOperator();
        }

        Value = newOperator;
    }

    private static string GetSpecifiedOperator()
    {
        var operatorId = RandomService.RandomInt(Bound);

        return operatorId switch
        {
            0 => " == ",
            1 => " != ",
            2 => " > ",
            3 => " < ",
            4 => " >= ",
            5 => " <= ",
            _ => throw new InvalidOperationException()
        };
    }

    public CompareOperator(Node parentNode) : base(parentNode, "CompareOperator")
    {
        Value = GetSpecifiedOperator();
    }

    public CompareOperator(Node parentNode, IList<Token> tokens) : base(parentNode, tokens)
    {
    }
}